test/join.jl

```diff
@@ -346,8 +343,8 @@ module TestJoin
         @test levels(join(B, A, on=:b, kind=:inner)[:b]) == ["a", "b", "c"]
         @test levels(join(A, B, on=:b, kind=:left)[:b]) == ["d", "c", "b", "a"]
         @test levels(join(B, A, on=:b, kind=:left)[:b]) == ["a", "b", "c"]
-        @test levels(join(A, B, on=:b, kind=:right)[:b]) == ["d", "c", "b", "a"]
-        @test levels(join(B, A, on=:b, kind=:right)[:b]) == ["a", "b", "d", "c"]
+        @test levels(join(A, B, on=:b, kind=:right)[:b]) == ["a", "b", "c"]
```

Comment:
Why does this change? The docstring for `join` gives guarantees about the levels, so either it wasn't correct before or it might need to be updated.

Reply:
In this PR the types and categorical levels of the right on-columns have priority over the left ones.
I think it's a logical thing. Also it's easier to implement.
So I can either patch the PR to the previous behaviour or update the docstring.

Reply:
As long as it's not too hard to implement, I'd choose the rule which is the simplest to explain in the docstring, which is a good indication that it's easy to understand. IIUC, the new behavior you propose is to give priority to the left column, except for `kind=:right`? I'm not too fan of exceptions, even if I see the justification.

Reply:
I've reverted to the old category levels ordering.